Gems/LyShine/Code/Source/UiTextInputComponent.cpp

```diff
@@ -108,6 +108,7 @@ namespace
         utf8String.erase(left, right - left);
     }
 
+#if !(defined(CARBONATED) && defined(AZ_PLATFORM_IOS))
```

Comment:
Why was this removed?

Reply:
Because it was causing an unused function error. I've removed the preprocessor defines

Reply:
That's odd.  We have many unused functions that don't throw errors.  Remove this guard to invoke the error and let's look at it together.

Reply:
I no longer get the error after removing all the AZ_PLATFORM_IOS defines